### PR TITLE
Fix incorrect data generation for simulated noise

### PR DESCRIPTION
**Pull request by spxiwh**
@SergeiOssokine reported some strange behaviour in the `hwinj` code where it would inject an injection at the wrong time.

This was diagnosed to be an issue in how pad-data is handled after the recent consolidation of code in the strain module. In particular at the moment if somebody uses any of the simulated data options, and does not explicitly set `pad-data` to 0, they will not get the correct amount of data as the padded data will be removed even though it was never added!

I'm proposing to fix this by correctly adding pad-data into the simulated options. I think this *is* needed as one might still try to high-pass filter this data (even if it isn't really adding anything to do that). 

I'm also a bit confused by having the default pad-data be 8. I guess here the sane default for simulated data and real data disagrees! I might think about changing this to 0, but don't feel too strongly (simulating 16s we don't need and then cutting it isn't *too* bad).

I also fix a line in the `hwinj` code to ensure that even if it doesn't return the length of data requested, the hwinj still goes in the correct place.

### CODE REVIEW

**Comment by titodalcanton** [comment]
> I'm also a bit confused by having the default pad-data be 8.

I also had the same confusion recently, although I think a nonzero default padding is a safer default when dealing with real data (FWIW, in my experience a default padding of 1 s is sufficient to cover the low/high pass filters at least).

**Comment by ahnitz** [comment]
Can we add a unit test here?

**Review by ahnitz — APPROVED** [review]

**Comment by ahnitz** [comment]
@spxiwh I see no reason not to merge this unless you wanted to hold off for a unittest.

**Pull request merged**
